### ISSUE TITLE
error in the build of the docker container

### ISSUE THREAD

**Issue by changemenemo** [hexparrot/mineos-docker#5]
```
@docker:/media/DOCK/docker/mineos-docker/debian-jessie$ docker build . -t mineos
Sending build context to Docker daemon  5.12 kB
Step 1 : FROM debian:jessie
 ---> 031143c1c662
Step 2 : MAINTAINER William Dizon <wdchromium@gmail.com>
 ---> Using cache
 ---> cae961a2994b
Step 3 : RUN echo "deb http://http.debian.net/debian jessie-backports main" >> /etc/apt/sources.list
 ---> Using cache
 ---> 33ff29312e57
Step 4 : RUN apt-get update && apt-get install -y   supervisor   rdiff-backup   screen   openjdk-8-jre-headless   rsync   git   curl   rlwrap   && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 ---> Using cache
 ---> 718b224a033c
Step 5 : RUN curl https://deb.nodesource.com/node_4.x/pool/main/n/nodejs/nodejs_4.3.1-1nodesource1~jessie1_amd64.deb > node.deb  && dpkg -i node.deb  && rm node.deb
 ---> Using cache
 ---> 1d9b18516ff6
Step 6 : RUN mkdir /usr/games/minecraft   && cd /usr/games/minecraft   && git clone --depth=1 https://github.com/hexparrot/mineos-node.git .   && cp mineos.conf /etc/mineos.conf   && chmod +x webui.js mineos_console.js service.js
 ---> Using cache
 ---> 524d58ea9980
Step 7 : RUN cd /usr/games/minecraft   && apt-get update   && apt-get install -y build-essential   && npm install   && apt-get remove --purge -y build-essential   && apt-get autoremove -y   && apt-get clean   && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 ---> Using cache
 ---> 2e90766131d8
Step 8 : RUN cp /usr/games/minecraft/init/supervisor_conf /etc/supervisor/conf.d/mineos.conf
 ---> Using cache
 ---> e86f2b4204d3
Step 9 : CMD /usr/bin/supervisord -n -c /etc/supervisor/supervisord.conf
 ---> Using cache
 ---> f9d789782c32
Step 10 : COPY entrypoint.sh /entrypoint.sh
stat /var/lib/docker/aufs/mnt/e895447807c904e256441a34b9847ae346548324822eed545f65b8a8d67a427d/entrypoint.sh: no such file or directory
```

**Comment by changemenemo** [comment]
seems okey if you change the /entrypoint in /something/entrypoint  like /usr/games/entrypoint

**Comment by hexparrot** [comment]
Is your repo definitely unchanged from the github version (no modifications)?

Here's what happens when I build with zero intermittent layers built: http://pastebin.com/wiKxEzyh

Here's what I get when I build it a second time, using the layer cache:

```
root@willonubuntu:~/mineos-docker/debian-jessie# docker build -t mineos2 .
Sending build context to Docker daemon  5.12 kB
Step 1 : FROM debian:jessie
 ---> a24c3183e910
Step 2 : MAINTAINER William Dizon <wdchromium@gmail.com>
 ---> Using cache
 ---> a5eb70d84799
Step 3 : RUN echo "deb http://http.debian.net/debian jessie-backports main" >> /etc/apt/sources.list
 ---> Using cache
 ---> 2383ae5467e6
Step 4 : RUN apt-get update && apt-get install -y   supervisor   rdiff-backup   screen   openjdk-8-jre-headless   rsync   git   curl   rlwrap   && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 ---> Using cache
 ---> cacdc5f7fe4e
Step 5 : RUN curl https://deb.nodesource.com/node_4.x/pool/main/n/nodejs/nodejs_4.3.1-1nodesource1~jessie1_amd64.deb > node.deb  && dpkg -i node.deb  && rm node.deb
 ---> Using cache
 ---> 8fff1ecb664d
Step 6 : RUN mkdir /usr/games/minecraft   && cd /usr/games/minecraft   && git clone --depth=1 https://github.com/hexparrot/mineos-node.git .   && cp mineos.conf /etc/mineos.conf   && chmod +x webui.js mineos_console.js service.js
 ---> Using cache
 ---> 3772c1bde635
Step 7 : RUN cd /usr/games/minecraft   && apt-get update   && apt-get install -y build-essential   && npm install   && apt-get remove --purge -y build-essential   && apt-get autoremove -y   && apt-get clean   && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 ---> Using cache
 ---> c33a0f3300dd
Step 8 : RUN cp /usr/games/minecraft/init/supervisor_conf /etc/supervisor/conf.d/mineos.conf
 ---> Using cache
 ---> b7c05835a68c
Step 9 : CMD /usr/bin/supervisord -n -c /etc/supervisor/supervisord.conf
 ---> Using cache
 ---> 7d3816b60650
Step 10 : COPY entrypoint.sh /entrypoint.sh
 ---> Using cache
 ---> c5e146b0b3bc
Step 11 : ENTRYPOINT /entrypoint.sh
 ---> Using cache
 ---> 580d270c36a1
Step 12 : EXPOSE 8443 25565-25570
 ---> Using cache
 ---> 72c60de236d2
Step 13 : VOLUME /var/games/minecraft
 ---> Using cache
 ---> 92ef40979152
Successfully built 92ef40979152
root@willonubuntu:~/mineos-docker/debian-jessie#
```

I've never actually built anything as a non-root user, but I see that you're using an unprivileged user.  Do you get the same issue when run as `root`?

**Comment by changemenemo** [comment]
I didn't use root because docker recommend docker group which should be the usual user because of security risk. And honestly, we, us ,you should know that we don't use root only for  specific usage like RAID managing or hardware managing, or the concept of sudo, blocking of ssh for root, everything that devs put in place to protect the system doesn't have any use.
The repo, I promise you, is unchanged from your github. It was the second time I build it since the first time it has to 
the layer cache? what do you mean by layer cache? Anyway you can see that it stops at step 10. So certainly it's because of you way to put entrypoint at roots. Anyway my solution works so it certainly because of the root problem.

**Comment by changemenemo** [comment]
what do you see here : [https://docs.docker.com/engine/reference/commandline/build/](ressources)

it's $ before docker not a # so what do you conclude?

**Comment by hexparrot** [comment]
> Anyway you can see that it stops at step 10. So certainly it's because of you way to put entrypoint at roots. Anyway my solution works so it certainly because of the root problem.

Anyway you can see that it doesn't stop for me at step 10.  So certainly it works, unchanged, through the two attempts that I built the docker image from the docker file.

First, from a completely clean install which has no cached layers to work from.  In other words, it had to download everything, hence why it's 1600 lines long and includes all the downloading and compiling of modules.

Then, the second attempt clearly indicates "using cache" and still completes.  So what do you conclude?

```
Step 10 : COPY entrypoint.sh /entrypoint.sh
 ---> c5e146b0b3bc
Removing intermediate container 755a66889ac3
Step 11 : ENTRYPOINT /entrypoint.sh
 ---> Running in 0024ae5c14e1
 ---> 580d270c36a1
Removing intermediate container 0024ae5c14e1
```

**Comment by changemenemo** [comment]
That like you said, you use root... And we shouldn't use root... So what? because of you using root, as you shouldn't need to as stated in the docs, everyone has to? 

Successfully built 92ef40979152
root@willonubuntu:~/mineos-docker/debian-jessie#

**Comment by changemenemo** [comment]
It's like for the cron jobs for the update of the UI. You don't want any help or modification. I even give you the solution and still you refuse it... that's unbelievable.

**Comment by changemenemo** [comment]
You know, if you would feel better, you can insult me if it gives you a better feeling about this, maybe it's me who is a stupid guy to follow java and C++ convention programming or docs recommandation for the platform which I'm using.

**Comment by hexparrot** [comment]
> And we shouldn't use root

What is this based on?  

This [docker issue](https://github.com/docker/docker/issues/1034) where they rule that docker should be used by root users only? 

Look it up, more sites anywhere will say that only root should be using docker than non-root users that should be using docker.

> I even give you the solution and still you refuse it... that's unbelievable.

You're not giving a solution.  I've shown you that I'm following a good--if not a best practice--and I'm able to accomplish the task without your modifications, which I rule to be thus-far not a change that improves anything.

> maybe it's me who is a stupid guy to follow java and C++ convention programming or docs recommandation for the platform which I'm using.

Feel free to cite a source that actually indicates it's better to use non-root users than root users.  I'll tell you one thing: showing the prompt on the docs is not a good supporting argument.

**Comment by changemenemo** [comment]
2013.... so please don't evolve because it will certainly hurts people? That's really your stand?

So showing the prompt is not a good argument mmmh right, so maybe this thing of non-rooted user is still in beta maybe ^^

http://askubuntu.com/questions/16178/why-is-it-bad-to-login-as-root
http://askubuntu.com/questions/477551/how-can-i-use-docker-without-sudo
http://www.infoworld.com/article/3030558/application-virtualization/docker-goes-rootless-and-thats-a-good-thing.html

just read that and tell me what you think okey?

**Comment by changemenemo** [comment]
look at the dates in the same time ;)

**Comment by hexparrot** [comment]
So, to recap:

> Users who can run Docker commands have effective root control of the system. Only grant this privilege to trusted users.

From [Oracle docs](http://docs.oracle.com/cd/E52668_01/E75728/html/section_rdz_hmw_2q.html), which decided that the default position is root-only.

> From a security perspective, Red Hat has expressed concerns with enabling access to the docker daemon from non-root users, absent auditing and proper logging.

Regarding CentOS, Fedora and RHEL [not letting docker run as a non-root user](https://www.projectatomic.io/blog/2015/08/why-we-dont-let-non-root-users-run-docker-in-centos-fedora-or-rhel/).

What they offer instead as a solution is instead to use `sudo`.

**Comment by changemenemo** [comment]
So your stand it's even, if the devs from docker have done some major rewrite of their code to use the docker group we shouldn't use it, taht's right?

**Comment by hexparrot** [comment]
It seems apparent you didn't actually read the links you gave:

http://askubuntu.com/questions/16178/why-is-it-bad-to-login-as-root

Relevant point: is not actually about docker at all.  wow.  It's about whether or not you should log into the system as root, which is not even slightly the same as what I'm recommending, which is ELEVATING TO ROOT, or using `sudo` to get the permissions necessary to run the docker commands.

http://askubuntu.com/questions/477551/how-can-i-use-docker-without-sudo

Relevant point: 

> Warning: The docker group (or the group specified with -G) is root-equivalent; see Docker Daemon Attack Surface details.
> 
>  The docker daemon must always run as the root user, but if you run the docker client as a user in the docker group **then you don't need to add sudo to all the client commands**.

So basically, this whole help is so that you "dont have to add sudo"--a convenience measure, _not_ a security measure.

**Comment by hexparrot** [comment]
You have brought attitude and toxicity to this issue and your involvement in this project provides no meaningful benefit to me or to the users who would use docker.

Your insistence that things need to change so you "don't need to sudo" is not a valid approach and honestly, I'm through with your contentious attitude. Have a nice day.
